src/scripts/content/todoist.js

```diff
@@ -3,9 +3,84 @@
 
 'use strict';
 
+function getProjectNameFromLabel(elem) {
+  var projectLabel = '', projectLabelEle = $('.pname', elem.parentNode.parentNode);
+  if (projectLabelEle) {
+    projectLabel = projectLabelEle.textContent.trim();
+  }
+  return projectLabel;
+}
+
+var levelPattern = /(?:^|\s)indent_([0-9]*?)(?:\s|$)/;
+function getParentEle(sidebarEle) {
+  var curLevel, parentClass, parentCandidate;
+  curLevel = sidebarEle.className.match(levelPattern)[1];
+  parentClass = 'indent_' + (curLevel - 1);
+
+  parentCandidate = sidebarEle;
+  while (parentCandidate.previousElementSibling) {
+    parentCandidate = parentCandidate.previousElementSibling;
+    if (parentCandidate.classList.contains(parentClass)) {
+      break;
+    }
+  }
+  return parentCandidate;
+}
+
+function isTopLevelProject(sidebarEle) {
+  return sidebarEle.classList.contains('indent_1');
+}
+
+function getProjectNameHierarchy(sidebarEle) {
+  var parentProjectEle, projectName;
+  projectName = $('.name', sidebarEle).firstChild.textContent.trim();
+  if (isTopLevelProject(sidebarEle)) {
+    return [projectName];
+  }
+  parentProjectEle = getParentEle(sidebarEle);
+  return [projectName].concat(getProjectNameHierarchy(parentProjectEle));
+}
+
+function projectWasJustCreated(projectId) {
+  return projectId.startsWith('_');
+}
+
+function getSidebarEle(elem) {
```

Comment:
I think a more accurate name for this would be `getSidebarCurrentEle` or something as it's looking for the current project item not the whole sidebar. Got me confused for a second. (also the var `sidebarEle`)